server/polar/integrations/github/service/issue.py

```diff
@@ -553,154 +549,55 @@ async def sync_external_org_with_repo_and_issue(
         issue_number: int,
     ) -> Tuple[Organization, Repository, Issue]:
         log.info(
-            "syncing external",
+            "syncing external issue",
             org_name=org_name,
             repo_name=repo_name,
             issue_number=issue_number,
         )
 
-        organization = await github_organization.get_by_name(
-            session, Platforms.github, org_name
-        )
-
-        if not organization:
-            log.info("organization not found by name", org_name=org_name)
-
-            try:
-                repo_response = await client.rest.repos.async_get(org_name, repo_name)
-                github_repo = repo_response.parsed_data
-                owner = github_repo.owner
-                is_personal = owner.type.lower() == "user"
-            except RequestFailed as e:
-                if e.response.status_code == 404:
-                    raise ResourceNotFound()
-                if e.response.status_code == 401:
-                    raise ResourceNotFound()
-                # re-raise other status codes
-                raise e
-
-            # check if we have org with same external_id
-            organization = await github_organization.get_by_external_id(
-                session, owner.id
+        try:
+            github_repository_response = await client.rest.repos.async_get(
```

Comment:
Previously this function would use locally cached results by default, and would only fallback to syncing from GitHub if it was missing. Maybe the method name is to blame here for the confusion, but that would be a nice property to keep. :)

That's especially important when called with `client=get_polar_client()` as we don't have too many API credits for that user.

Reply:
Will improve that!